screen.go

```diff
@@ -141,13 +141,21 @@ func (s *screen) applyEscape(code rune, instructions []string) {
 				// This line should be equivalent to K0
 				s.clear(s.y, s.x, screenEndOfLine)
 				// Truncate the screen below the current line
-				s.screen = s.screen[:s.y+1]
+				thisLineOrEndOfScreen := len(s.screen)
+				if s.y+1 <= thisLineOrEndOfScreen {
+					thisLineOrEndOfScreen = s.y+1
+				}
+				s.screen = s.screen[:thisLineOrEndOfScreen]
```

Comment:
I wonder if this could be a no-op when there are no lines below:

```suggestion
				if len(s.screen) > s.y {
					s.screen = s.screen[:s.y+1]
				}
```

(and the same for the inverse operation)

Reply:
Good call, I'll look at optimising that.